File: lab2no3.java
import java.util.Scanner;

public class Main {
    
    public static boolean checkTimeFormat(int num) {
        if (num < 100) {
            return false;
        }
        int hours = num / 100;
        int minutes = num % 100;
        
        if (hours >= 0 && hours <= 23 && minutes >= 0 && minutes <= 59) {
            return true;
        } else {
            return false;
        }
    }
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int count = scanner.nextInt();
        
        for (int i = 0; i < count; i++) {
            int num = scanner.nextInt();
            if (num == -1) {
                break;
            }
            if (checkTimeFormat(num)) {
                System.out.println("YES");
            } else {
                System.out.println("NO");
            }
        }
        
        scanner.close();
    }
}

File: lab2no3.kt
import java.util.Scanner

fun checkTimeFormat(num: Int): Boolean {
    if (num < 100) {
        return false
    }
    val hours = num / 100
    val minutes = num % 100

    return hours in 0..23 && minutes in 0..59
}

fun main() {
    val scanner = Scanner(System.in)
    print("Enter the number of test cases: ")
    val caseCount = scanner.nextInt()
    var i = 0

    while (i < caseCount) {
        print("Enter time format (hhmm) or -1 to exit: ")
        val num = scanner.nextInt()
        if (num == -1) {
            break
        }
        if (checkTimeFormat(num)) {
            println("YES")
        } else {
            println("NO")
        }
        i++
    }
}
